src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt

```diff
@@ -198,20 +239,34 @@ class CargoTestEventsConverter(
         suitesStack.add(suite)
     }
 
+    private enum class State {
+        ReceivesStartMessage,
+        ReceivesExecutableName,
+        ReceivesDoctestPackageName
+    }
+
     companion object {
-        private val ESCAPED_SEPARATOR: String = File.separator.replace("\\", "\\\\")
-        private val TARGET_PATH_PART_REGEX: Regex = Regex("[ $ESCAPED_SEPARATOR]target$ESCAPED_SEPARATOR")
+        private val PARSER: JsonParser = JsonParser()
         private const val ROOT_SUITE: String = "0"
         private const val NAME_SEPARATOR: String = "::"
+        private const val DOCTESTS_SUFFIX = "doctests"
 
-        private val COMPARISON_ASSERT_REGEX: Regex =
-            Regex("""thread '.*' panicked at 'assertion failed: `\(left == right\)`\s*left: `(.*)`,\s*right: `(.*?)`(:.*)?'""")
+        @Language("RegExp")
+        private val DOCTEST_PATH_RE: Regex = """"[0-9 a-z_A-Z\-\\.]+ - """.toRegex()
```

Comment:
Does it intentionally has `"` at the start of regexp?

Reply:
Yes.
```
{ "type": "test", "event": "started", "name": "src/lib.rs - foo (line 9)" }
                                              ↑
```